### PR TITLE
App: add option to material combobox to restore to original values from file

### PR DESCRIPTION
**Pull request by prewettg**
This change is [Reviewable](https://reviewable.io/reviews/intel-isl/open3d/1873)

### CODE REVIEW

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5560990&bid=18236788)

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5560990&bid=18236788)

File: src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp [hunk 1]
```diff
@@ -1636,6 +1674,20 @@ void GuiVisualizer::SetGeometry(
         impl_->settings_.wgt_point_size->SetEnabled(num_point_clouds > 0);
     }
 
+    if (!impl_->settings_.loaded_materials_.empty()) {
+        int resetIdx = impl_->settings_.wgt_prefab_material->AddItem(
+                MATERIAL_FROM_FILE_NAME.c_str());
+        impl_->settings_.wgt_prefab_material->SetSelectedIndex(resetIdx);
+        ;
```
Comment (pvarvak) [inline src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp:1681]: remove empty ;?

File: src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp [hunk 2]
```diff
@@ -1636,6 +1674,20 @@ void GuiVisualizer::SetGeometry(
...
+        impl_->settings_.wgt_prefab_material->ChangeItem(
+                (DEFAULT_MATERIAL_NAME + " [default]").c_str(),
+                DEFAULT_MATERIAL_NAME.c_str());
+    } else {
+        impl_->settings_.wgt_prefab_material->ChangeItem(
+                DEFAULT_MATERIAL_NAME.c_str(),
+                (DEFAULT_MATERIAL_NAME + " [default]").c_str());
```
Comment (pvarvak) [inline src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp:1688]: move this (identical?) statement out of if and else and do it after for both of them?

File: src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp [hunk 3]
```diff
@@ -842,23 +852,50 @@ struct GuiVisualizer::Impl {
...
-        auto prefab_it = prefab_materials_.find(name);
-        if (prefab_it != prefab_materials_.end()) {
-            auto &prefab = prefab_it->second;
-            if (!settings_.user_has_changed_color) {
-                settings_.current_materials.lit.base_color = prefab.base_color;
-                settings_.wgt_material_color->SetValue(prefab.base_color.x(),
-                                                       prefab.base_color.y(),
-                                                       prefab.base_color.z());
+        if (name == MATERIAL_FROM_FILE_NAME) {
+            settings_.user_has_changed_color = false;
```
Comment (pvarvak) [inline src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp:856]: we do this here and after the for loop, does the for loop change it?

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5560990&bid=18236788)

File: src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp (hunk 2 above)
Comment (prewettg) [reply src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp:1695]: They aren't identical. One changes from "Default text" -> "Default text [default]" and the other does the reverse.

File: src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp (hunk 1 above)
Comment (prewettg) [reply src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp]: This crappy butterfly keyboard is driving me nuts. I think my enter key repeated itself. Removed.

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5560990&bid=18236788)

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5560990&bid=18236788)

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=5560990&bid=18236788)

**Pull request merged**